airflow/jobs/scheduler_job.py

```diff
@@ -1483,10 +1487,23 @@ def _do_scheduling(self, session) -> int:
             session.expunge_all()
             # END: schedule TIs
 
-            # TODO[HA]: Do we need to call
-            # _change_state_for_tis_without_dagrun (2x) that we were before
-            # to tidy up manually tweaked TIs. Do we need to do it every
-            # time?
+            # TODO[HA]: Do we need to do it every time?
+            self._change_state_for_tis_without_dagrun(
+                dag_ids=self.dagbag.dag_ids,
```

Comment:
At this point, this is basically ALL dags, isn't it? So we could change the logic to not take dag IDs at all.

Reply:
Updated: https://github.com/astronomer/airflow/pull/956/commits/6dd35bcc3986429ebed15b7287fc204a7d4fdcb0